Lib/test/test_asyncio/test_futures2.py

```diff
@@ -1,13 +1,31 @@
 # IsolatedAsyncioTestCase based tests
 import asyncio
 import unittest
+import traceback
 
 
 def tearDownModule():
     asyncio.set_event_loop_policy(None)
 
 
 class FutureTests(unittest.IsolatedAsyncioTestCase):
+
```

Comment:
I'm assuming this test runs both the C and the python versions?

Reply:
I think there's a separate run of the tests that disables `_asyncio`? I have no idea where this is done though.

Reply:
If I revert the C fix the test fails. If I revert the python fix it doesn't.

Reply:
Yeah, you have to use a pattern that creates two separate test classes, similar to [this](https://github.com/python/cpython/blob/14fea6b4d25658bc00adbb97dd40ea3d3e6843ad/Lib/test/test_asyncio/test_futures.py#L638-L667) (but you don't need to test with a Python subclass of `CFuture`).

Reply:
Fixed in https://github.com/python/cpython/pull/30274/commits/9d6bae0b281e764a8f83f2e91b9dfff37588917c